src/tactic/squeeze.lean

```diff
@@ -178,25 +200,14 @@ do v ← target >>= mk_meta_var,
      instantiate_mvars g },
    let vs := g.list_constant,
    vs ← vs.mfilter is_simp_lemma,
-   vs ← vs.mmap strip_prefix,
```

Comment:
What was the call to `strip_prefix` doing before? This makes me suspect that the new behavior might leave unnecessary prefixes in, where the old one didn't, is that right?

Reply:
from the tests, it doesn't seem so; I'm glad to put it back if necessary but I haven't seen a situation that needs it. Should I leave a comment that it used to be here, and to try add it back if needed?

Reply:
It makes a difference in the following situation:
```lean
open nat

attribute [simp] nat.mul_succ 

example (n m : ℕ) : n * m.succ = n*m + n :=
by squeeze_simp
```
Without the `strip_prefix` line, `squeeze_simp` suggests `nat.mul_succ`; with it, the suggestion is `mul_succ`.

I'm not sure if this conflicts with your changes. But I'd rather not remove it unless we really have to, since it will make the output of `squeeze_simp` needlessly verbose. Based on the logic of your code I think we probably don't need to delete it.

Reply:
apologies for the messed up merge; I fixed this and added a test for it, too.